tests/util/test_yaml.py

```diff
@@ -90,26 +93,33 @@ def test_include_dir_list(self, mock_walk):
     def test_include_dir_list_recursive(self, mock_walk):
         """Test include dir recursive list yaml."""
         mock_walk.return_value = [
-            ['/tmp', ['tmp2'], ['zero.yaml']],
+            ['/tmp', ['tmp2', '.ignore', 'ignore'], ['zero.yaml']],
             ['/tmp/tmp2', [], ['one.yaml', 'two.yaml']],
+            ['/tmp/.ignore', [], []],
+            ['/tmp/ignore', [], ['.ignore.yaml']]
         ]
 
         with patch_yaml_files({
-                '/tmp/zero.yaml': 'zero', '/tmp/tmp2/one.yaml': 'one',
-                '/tmp/tmp2/two.yaml': 'two'
+            '/tmp/zero.yaml': 'zero',
+            '/tmp/tmp2/one.yaml': 'one',
+            '/tmp/tmp2/two.yaml': 'two'
         }):
             conf = "key: !include_dir_list /tmp"
             with io.StringIO(conf) as file:
                 doc = yaml.yaml.safe_load(file)
+                assert '.ignore' not in mock_walk.return_var[0].dirs
```

Comment:
Just before `safe_load`, line 109, can you make sure it really is there (ensure that check at the correct place):

```python
assert '.ignore' in mock_walk.return_var[0].dirs, "Expecting .ignore in here"
```

For some reason I suspect it might be `return_value[0][1]`, and not sure about the `return_value[0].dirs` anymore.